### PR TITLE
Add more options to action

### PR DESCRIPTION
**Pull request by dagwieers**
This PR includes:
- Added `args` option to add additional sonar-scanner arguments
- Merge Dockerfile LABELs to speed up processing Dockerfile
- Update documentation

By adding these the sonar-scanner can run without a sonar-project.properties file.

### CODE REVIEW

**Comment by hakimio** [comment]
Would be quite useful for us too if this could be merged. 
@simonbrandhof What do you think about it?

**Comment by hakimio** [comment]
@dagwieers you need to fix merge conflicts with master branch.

**Comment by dagwieers** [comment]
@hakimio Done.

**Comment by dagwieers** [comment]
IMO it is better to add as many options that are useful to end-users, rather than use the **args** option. But if you configure the sonarcloud service properly, most of these properties are configured via the web interface AFAICT. So this is the bare minimum for a working setup.

**Comment by hakimio** [comment]
I would vote for leaving *args* option since I personally need some extra arguments like the following:
-  sonar.javascript.lcov.reportPaths
- sonar.testExecutionReportPaths
- sonar.tests
- sonar.test.inclusions

**Comment by dagwieers** [comment]
@hakimio These options can be configured in the Sonarcloud web interface as well.
Screenshot from 2020-03-31 11-12-35

**Comment by hakimio** [comment]
I know that I can configure test.inclusions through web interface but I don't know where to find the other two.

**Comment by dagwieers** [comment]
I could only find one of those two under the **Languages**. It is true that the web interface is far from intuitive. I could think of better ways to represent the configuration options.

Screenshot from 2020-03-31 11-40-06

**Comment by dagwieers** [comment]
What is needed to get this reviewed and merged?

**Comment by hakimio** [comment]
@simonbrandhof ?

**Comment by dagwieers** [comment]
Another attempt to get this merged! Anyone?

**Comment by dagwieers** [comment]
@simonbrandhof @hakimio This GitHub action is failing because a certificate of **binaries.sonarsource.com** has expired. CI is now failing for all our projects :-(

```
Resolving binaries.sonarsource.com (binaries.sonarsource.com)... 91.134.125.245
Connecting to binaries.sonarsource.com (binaries.sonarsource.com)|91.134.125.245|:443... connected.
ERROR: cannot verify binaries.sonarsource.com's certificate, issued by ‘/C=FR/ST=Paris/L=Paris/O=Gandi/CN=Gandi Standard SSL CA 2’:
  Issued certificate has expired.
To connect to binaries.sonarsource.com insecurely, use `--no-check-certificate'.
```

**Comment by dagwieers** [comment]
So anyone having problems due to the expired certificate on binaries.sonarcloud.com can use **dagwieers/sonarcloud-github-action@more-options** as an alternative. This branch has cert validation disabled.


```yaml
    - name: Analyze with SonarCloud
      uses: dagwieers/sonarcloud-github-action@more-options
      with:
        organization: add-ons
        projectKey: add-ons_plugin.video.vrt.nu
      env:
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
      continue-on-error: true
```

**Comment by dagwieers** [comment]
Now **binaries.sonarsource.com** seems to be down, which causes automated testing to fail. What is worse, the workflow is not even attempted as it bails out at the very start. So there is no means to make sonarcloud optional.

**Comment by simonbrandhof** [comment]
> Now **binaries.sonarsource.com** seems to be down, which causes automated testing to fail. 

This is fixed in v1.3 by removing the dependency on binaries.sonarsource.com.

**Comment by simonbrandhof** [comment]
Regarding the initial proposal, I think that splitting the settings into two types (native arguments declared in `with`+ additional arguments declared in `args`) does not improve user experience. It hides the mapping with the official documented parameters that are prefixed with `sonar.`. 
Moreover please note that the organization and project keys may be auto-guessed someday and will remove the need for declaring them.

**Comment by hakimio** [comment]
@simonbrandhof what's your solution for mono-repos which have multiple Sonar Cloud projects and multiple github Sonar Cloud workflows?

**Comment by simonbrandhof** [comment]
> @simonbrandhof what's your solution for mono-repos which have multiple Sonar Cloud projects and multiple github Sonar Cloud workflows?

That does not relate to this thread. I suggest to have a look at [MMF-2036](https://jira.sonarsource.com/browse/MMF-2036), maybe [MMF-2001](https://jira.sonarsource.com/browse/MMF-2001), and to use the [community forum](https://community.sonarsource.com/tag/sonarcloud) to provide feedback or request for help. Thanks.

**Comment by dagwieers** [comment]
> This is fixed in v1.3 by removing the dependency on binaries.sonarsource.com.

Oh great, I totally missed a new release. I updated this branch.

**Comment by dagwieers** [comment]
> Regarding the initial proposal, I think that splitting the settings into two types (native arguments declared in with+ additional arguments declared in args) does not improve user experience. It hides the mapping with the official documented parameters that are prefixed with sonar..

Agreed, I have removed the `args` parameter.

**Comment by hakimio** [comment]
@dagwieers I am pretty sure @simonbrandhof is saying that he doesn't want this merged at all. So, I don't think you should bother with the PR anymore.
And having `args` for additional parameters was quite useful at least for us like I've explained above.

**Comment by dagwieers** [comment]
@hakimio That is not what @simonbrandhof said. So if I have to choose beween not being merged, or merged partial, I go for the second option ;-)

Another option is to use `solar_` the prefix. Would you prefer that? (I will have to update all my projects, but am willing to do so if that would be a solution for you)

**Comment by simonbrandhof** [comment]
Hi guys. Indeed it seems it deserves clarification!
I'm in favor of keeping exactly the `sonar.*` parameters as documented and supported in other tools/CI vendors. Said differently, only `args` should be implemented. This improvement brings already value because it removes the need for creating the file `sonar-project.properties`.

**Comment by dagwieers** [comment]
@simonbrandhof That is unfortunate because I prefer this as an interface over putting everything as command-options. But functionally `args` allows everything I need, so I will update to only support args.

Thanks for clarifying !

**Comment by dagwieers** [comment]
Implemented as requested !

File: README.md [hunk 1]
```diff
@@ -59,6 +59,26 @@ with:
...
+      -Dsonar.projectKey=my-projectkey
+      -Dsonar.python.coverage.reportPaths=coverage.xml
+      -Dsonar.sources=lib/
+      -Dsonar.test.exclusions=tests/**
+      -Dsonar.tests=tests/
+      -Dsonar.verbose=true
+```
+
+More information about possible sonar-scanner parameters is found in the documentation at:
+https://docs.sonarqube.org/latest/analysis/analysis-parameters/
```
Comment (simonbrandhof) [inline README.md:80]: Should be replaced by https://sonarcloud.io/documentation/analysis/analysis-parameters/

File: README.md [hunk 2]
```diff
@@ -59,6 +59,26 @@ with:
...
+      -Dsonar.organization=my-organization
+      -Dsonar.projectKey=my-projectkey
+      -Dsonar.python.coverage.reportPaths=coverage.xml
+      -Dsonar.sources=lib/
+      -Dsonar.test.exclusions=tests/**
+      -Dsonar.tests=tests/
+      -Dsonar.verbose=true
+```
+
+More information about possible sonar-scanner parameters is found in the documentation at:
```
Comment (simonbrandhof) [inline README.md:79]: I suggest to remove the term "sonar-scanner" for the sake of simplicity.

File: README.md (hunk 1 above)
Comment (dagwieers) [reply README.md]: Fixed.

File: README.md (hunk 2 above)
Comment (dagwieers) [reply README.md]: Very well. Implemented as requested.

**Comment by dagwieers** [comment]
@simonbrandhof If you could re-review and merge as soon as you can, so we can fix our GitHub workflows. I would rather use the official branch with this included, then fix all our projects two times. (First using my branch with the new options, then again with an official release) Thanks!

**Comment by dagwieers** [comment]
@simonbrandhof Can this be merged?

**Comment by tophercf** [comment]
other actions using sonar scanner have the args option; this is useful for dynamic inputs coming from the CI provider, like bas, target branch name, and PR index value for scans related to PR requests.  Echoing @dagwieers; would definitely like to see this merged, to allow for all available params to be available for use in the action.

PR dynamic properties that need to be set (pulled from CI environment vars at run time): https://docs.sonarqube.org/7.7/analysis/pull-request/
example actions from other sonar scan implementors: https://github.com/Warchant/setup-sonar-scanner

**Review by janos-ss — APPROVED** [review]

**Review by simonbrandhof — APPROVED** [review]

**Comment by simonbrandhof** [comment]
> other actions using sonar scanner have the args option; this is useful for dynamic inputs coming from the CI provider, like base branch name, target branch name, and PR index value for scans related to PR requests. Echoing @dagwieers; would definitely like to see this merged, to allow for all available params to be available for use in the action.
> 
> PR dynamic properties that need to be set (pulled from CI environment vars at run time): https://docs.sonarqube.org/latest/analysis/pull-request/
> example actions from other sonar scan implementors: https://github.com/Warchant/setup-sonar-scanner

The settings related to branches and pull requests are already guessed on GitHub Action environment. No need to configure anything. Note that this auto-configuration feature is implemented directly in the scanner, not in this GitHub Action.

**Comment by jkowalleck** [comment]
@janos-ss  is this version ready to be tagged?

**Comment by simonbrandhof** [comment]
Done. It's `v1.4`.

**Pull request merged**